ipld/read.go

```diff
@@ -26,21 +26,57 @@ func RetrieveData(
 	codec rsmt2d.Codec,
 ) (*rsmt2d.ExtendedDataSquare, error) {
 	edsWidth := len(dah.RowsRoots)
-	rowRoots := dah.RowsRoots
 	dataSquare := make([][]byte, edsWidth*edsWidth)
 
-	quadrant, err := pickRandomQuadrant(rowRoots)
+	q := rand.Intn(4) + 1 //nolint:gosec
+	quadrant, err := pickQuadrant(q, dah.RowsRoots)
```

Comment:
Maybe unrelated to this PR, but I see that it looks like you're only sampling from row roots here? This should be changed to sample from both rows and columns to be secure. Otherwise rows will be sampled more than columns.

Reply:
can make it within this PR. Randomising between rows/cols will not take much time I guess.